File: maximum-difference-between-adjacent-elements-in-a-circular-array.ts
{
    function maxAdjacentDistance(nums: number[]): number {
        let max = Math.abs(nums[0] - nums.at(-1));

        for (let i = 1; i < nums.length; i++) {
            const abs_sum = Math.abs(nums[i - 1] - nums[i]);

            max = Math.max(abs_sum, max);
        }

        return max;
    };
    console.log(maxAdjacentDistance([1,2,4]))
}

File: maximum-difference-between-adjacent-elements-in-a-circular-array.rs
pub fn max_adjacent_distance(nums: Vec<i32>) -> i32 {
    let n = nums.len();
    let mut ans = 0;
    for i in 0..n {
        let j = (i + 1) % n;
        let diff = (nums[i] - nums[j]).abs();
        if diff > ans {
            ans = diff;
        }
    }
    ans
}
